### ISSUE TITLE
Windows: R_HOME not used in situation.py if `r_home_from_registry()` returns None

### ISSUE THREAD

**Issue by paulmueller** [rpy2/rpy2#744]
**Describe the issue or bug**

situation.py does not use R_HOME on Windows, even if the environment variable is set. The problem is that if `r_home_from_registry()` returns None, ``iter_info`` returns without proceeding with R_HOME.

**To Reproduce**

Install miniconda and run ``conda install -c conda-force rpy2``. This will install R alongside rpy2, but not add anything to Path or Windows registry.

```python
python -c "from rpy2 import situation; [print(s) for s in situation.iter_info()]"
```
![situation1](https://user-images.githubusercontent.com/2853022/98563250-3a6e6f00-22ab-11eb-8550-b44b3a56c0d6.PNG)

**Expected behavior**

I would expect situation.py to use the R_HOME environment variable. If I modify situation.py, then I get the following output:
![situation2](https://user-images.githubusercontent.com/2853022/98563444-786b9300-22ab-11eb-94d0-b3eda98f54f5.PNG)

**Additional context**
I am currently trying to get rpy2 to run on Appveyor. This is just one of the issues I am encountering. I would be willing to create a PR for this.

**Comment by lgautier** [comment]
The module `situation.py` does not return much when used a script. It merely has the side effect of printing information in the terminal.

Otherwise the environment variable `R_HOME` seems to be fetched, returned by the iterator (and then printed in the terminal):
https://github.com/rpy2/rpy2/blob/master/rpy2/situation.py#L285

May be I am missing something?

**Comment by paulmueller** [comment]
This is not about actually "return"ing something. ``iter_info`` returns/exits in line 307 https://github.com/rpy2/rpy2/blob/master/rpy2/situation.py#L307
if ``r_home_default`` is None, independent of whether ``r_home`` is None or not.

That means if ``r_home_from_registry()`` returns None on Windows, then ``iter_info`` stops right there, even if ``r_home`` was given.

Yes, it prints ``R_HOME``, but it does not print anything else beyond that (rpy2 status, R version, additional directories, C extension).

**Comment by lgautier** [comment]
Ah, got it. It returns on line 307.

I can't remember why it is this why. I suspect that at the time this was written I was unsure about what should be the behavior in that case, so I made it spit a warning and exit.

**Comment by lgautier** [comment]
Fixed with revision d2796394bc2384fe1e662154084279327751b239.

This is in the dev branch (future rpy2 3.4.0). Let me know if a blocker on your end, and you'd like me to look into having it in a bugfix release in the 3.3.x series.

**Comment by paulmueller** [comment]
Thanks! No bugfix in 3.3.x necessary.
